ui/tests/unit/adapters/aws-credential-test.js

```diff
@@ -0,0 +1,78 @@
+import { moduleFor, test } from 'ember-qunit';
+import apiStub from 'vault/tests/helpers/noop-all-api-requests';
+import needs from 'vault/tests/unit/adapters/_adapter-needs';
+
+moduleFor('adapter:aws-credential', 'Unit | Adapter | aws credential', {
+  needs,
+  beforeEach() {
+    this.server = apiStub();
+  },
+  afterEach() {
+    this.server.shutdown();
+  },
+});
+
+const storeStub = {
+  pushPayload() {},
+  serializerFor() {
+    return {
+      serializeIntoHash() {},
+    };
+  },
+};
+
+let makeSnapshot = obj => {
+  obj.role = {
+    backend: 'aws',
+    name: 'foo',
+  };
+  obj.attr = attr => obj[attr];
+  return obj;
+};
+
+const type = {
+  modelName: 'aws-credential',
+};
+
+const cases = [
+  ['iam_user type', [storeStub, type, makeSnapshot({ credentialType: 'iam_user', ttl: '3h' })], 'GET'],
+  [
+    'federation_token type with ttl',
+    [storeStub, type, makeSnapshot({ credentialType: 'federation_token', ttl: '3h', roleArn: 'arn' })],
+    'POST',
+    { ttl: '3h' },
+  ],
+  [
+    'federation_token type no ttl',
+    [storeStub, type, makeSnapshot({ credentialType: 'federation_token', roleArn: 'arn' })],
+    'POST',
+  ],
+  [
+    'assumed_role type no arn, no ttl',
+    [storeStub, type, makeSnapshot({ credentialType: 'assumed_role' })],
+    'POST',
+  ],
+  [
+    'assumed_role type no arn',
+    [storeStub, type, makeSnapshot({ credentialType: 'assumed_role', ttl: '3h' })],
+    'POST',
+    { ttl: '3h' },
+  ],
+  [
+    'assumed_role type',
+    [storeStub, type, makeSnapshot({ credentialType: 'assumed_role', roleArn: 'arn', ttl: '3h' })],
+    'POST',
+    { ttl: '3h', role_arn: 'arn' },
+  ],
+];
+cases.forEach(([description, args, expectedMethod, expectedRequestBody]) => {
+  test(`aws-credential: ${description}`, function(assert) {
+    assert.expect(3);
+    let adapter = this.subject();
+    adapter.createRecord(...args);
+    let { method, url, requestBody } = this.server.handledRequests[0];
+    assert.equal(url, '/v1/aws/creds/foo', `calls the correct url`);
+    assert.equal(method, expectedMethod, `${description} uses the correct http verb: ${expectedMethod}`);
+    assert.equal(requestBody, JSON.stringify(expectedRequestBody));
+  });
+});
```

Comment:
Bit nitpicky this, but I've also noticed this elsewhere. Would you call this a unit test?

It seems to be testing more than a single isolated unit? It looks like this is testing `urlForCreateRecord` and also `createRecord`.

I've just had a quick snoop around the code here, plus a couple of different `ember-data` versions, and it looks like it's impossible to dynamically set the request method without using `ds-ajax-improved` so you have to overwrite the entire `createRecord` method, so I can see why you've done that here.

Could you overwrite `urlForCreateRecord` and then unit test that? Then you could stub `urlForCreateRecord` and `this.ajax` to unit test `createRecord`. Alternatively move this to an integration and/or acceptance test?

Reply:
Ah yeah fair, these are probably more accurately described as integration tests since they're testing both the adapter and the serializer usually. I'll probably change that after all of the ember updates get in since the 3.0 update rewrites most of the tests for the new style async/await tests: https://github.com/hashicorp/vault/pull/5342

Reply:
That's merged now, going to move them over in another PR!